src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -193,10 +199,13 @@ def login_with_managed_identity(self, resource, identity_id=None):
                 if not authenticated:
                     try:
                         # Try object ID
-                        # TODO: Support resource ID in Azure Identity
-                        credential = ManagedIdentityCredential(object_id=identity_id)
+                        credential = ManagedIdentityCredential(identity_config={"object_id": identity_id})
+                        credential.get_token(scope)
                         id_type = self.MANAGED_IDENTITY_OBJECT_ID
                         authenticated = True
+                    except ClientAuthenticationError as e:
+                        logger.debug('MSI authentication error: %s', e.message)
+                        logger.info('Sniff: not an MSI object id')
```

Comment:
Consider using more instructive word other than `Sniff`. Also `MSI` is a deprecated name. Let's use `Managed Identity`.